media/js/base/stub-attribution.js

```diff
@@ -290,11 +334,14 @@ if (typeof window.Mozilla === 'undefined') {
         // make the XHR request to the stub authentication service.
         } else if (!StubAttribution.isFirefoxNewScene2()) {
 
-            data = StubAttribution.getAttributionData();
+            // Wait for GA to load so that we can pass along visit ID.
+            StubAttribution.waitForGoogleAnalytics(function() {
```

Comment:
Does this mean that we'll only get Attribution if the user sticks around for over 4 seconds if they don't get GA? If so is it possible to collect the other attribution data more quickly and then tack on the visit_id whenever it becomes available?

Reply:
What do you mean by collecting other attribution data more quickly? Would that mean making 2 calls to the auth end point? 

If so, what should we do in that instance, throw away the previous data? Or is is possible to make a second call to encode only the visit_id and then add that to what we already have? It feels like this is a point where complexity might start to creep in…

Reply:
We would have to throw away the old data... hmm... I actually thought that the signing of the collected data happened later on the download page itself. I forgot that it actually happens immediately. If you think this won't result in much loss of attribution then I'm cool with it like it is. But if we do see that a lot of people don't have GA and we miss a lot of attribution as a result we should consider getting the data in two steps perhaps. It's also possible that the rest of this data is superfluous once we get the GA ID stuff working, so hopefully we can simplify everything after it's proved to be working.

Reply:
A more robust way to implement this imho would be for bedrock to generate the unique ID, which we then pass both into GA and stub attribution independently. But I’m not 100% clear on prior discussion for this feature or the other work to support it though, so if this is the way it needs to work the I’ll see if I can tweak things a little more.

Reply:
the whole idea though is that we want this to respect privacy, and the only way to do that is for this download ID to stay only in our private database. If it's in GA then it's more data that we're giving GA. All that we need to do is get the GA ID to bouncer if one exists. It then generates the download ID and stores them.

Reply:
It feels a bit like the same level of tracking is going on here regardless to where an ID gets generated. We’re still piecing together the same level of information about visitors. If this is the way it’s been designed though then let’s try and make it work 👍

Reply:
The point is to restrict access to the info required to track very tightly.

Reply:
I think we could probably still do that depending on how it was implemented, but this is probably another convo. I have another approach I might try here tomorrow to get us what we need.